### PR TITLE
ipn/ipnlocal: add identity headers to HTTP serve proxy

### PR DESCRIPTION
**Pull request by soniaappasamy**
Adds two new headers to HTTP serve proxy:
- `Tailscale-User-Login`: Filled with requester's login name.
- `Tailscale-User-Name`: Filled with requester's display name.

These headers only get filled when the SrcAddr is associated with a non-tagged (i.e. user-owned) node within the client's Tailnet.

The headers are passed through empty when the request originated from another tailnet, or the public internet (via funnel).
    
Updates https://github.com/tailscale/tailscale/issues/6954

### CODE REVIEW

File: ipn/ipnlocal/serve.go [hunk 1]
```diff
@@ -469,6 +466,37 @@ func (b *LocalBackend) proxyHandlerForBackend(backend string) (*httputil.Reverse
...
+		// No identity headers to add.
+		return
+	}
+	if !node.IsTagged() {
+		r.Out.Header.Set("X-Tailscale-User-Login", user.LoginName)
+		r.Out.Header.Set("X-Tailscale-User-Name", user.DisplayName)
+	} else {
+		r.Out.Header.Set("X-Tailscale-Tags", strings.Join(node.Tags, ","))
+	}
+	r.Out.Header.Set("X-Tailscale-Node", node.ComputedName)
```
Comment (bradfitz) [inline ipn/ipnlocal/serve.go:497]: We should drop the "X-": https://www.rfc-editor.org/rfc/rfc6648

Nobody will conflict with "Tailscale-" prefixed headers. :)

Comment (soniaappasamy) [reply ipn/ipnlocal/serve.go]: Updated to `X-Webauth-User` and `X-Webauth-Name` as more standard headers used by other services, such as grafana and gitea.

Comment (soniaappasamy) [reply ipn/ipnlocal/serve.go]: Caught up with @bradfitz on Slack. I'm opting to go with `Tailscale-User-Login` and `Tailscale-User-Name` headers, and will add a followup commit making these easily configurable against other services, such as grafana. This should still keep things easily plug-and-play with common services, but lets us avoid being tied to other service's concepts of "users"/"logins". Also will allow us to keep consistent `Tailscale-` namings for tagged identity/other fields in the future that may not have matching equivalents in other services.

cc @willnorris who had some thoughts on header names.

File: ipn/ipnlocal/serve.go [hunk 2]
```diff
@@ -398,14 +398,19 @@ func (b *LocalBackend) tcpHandlerForServe(dport uint16, srcAddr netip.AddrPort)
 	return nil
 }
 
+func getServeHTTPContext(r *http.Request) (c *serveHTTPContext, ok bool) {
+	c, ok = r.Context().Value(serveHTTPContextKey{}).(*serveHTTPContext)
+	return c, ok
```
Comment (bradfitz) [inline ipn/ipnlocal/serve.go:403]: two-three line functions is one of the few times naked returns are permissible if you prefer. either way.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**